doc/api/cli.md

```diff
@@ -595,6 +595,31 @@ Path to the file used to store the persistent REPL history. The default path is
 `~/.node_repl_history`, which is overridden by this variable. Setting the value
 to an empty string (`''` or `' '`) disables persistent REPL history.
 
+### `NODE_UNHANDLED_REJECTION=state`
+<!-- YAML
+added: REPLACEME
+-->
+
+Setting this environment variable allows fine grained control over the behavior
+of unhandled rejections. This may be set to either one of `SILENT`, `WARN`,
+`ERROR_ON_GC` or `ERROR` while the default behavior without using the
+environment variable is the same as `WARN`. `WARN` reflects the
+[`unhandledRejection hook`][].
+
+Setting the environment variable to `SILENT` prevents unhandled rejections
+from being logged. The [`unhandledRejection hook`][] is still triggered in case
+it is set.
+
+If set to `ERROR_ON_GC` unhandled rejections that are garbage collected are
+raised as uncaught exceptions.
```

Comment:
It seems that when enabling `--abort-on-uncaught-exception`, these exceptions do not make the process abort:

```
$ ulimit -c
unlimited
$ ../../node --expose-gc --abort-on-uncaught-exception promise_unhandled_reject.js
First GC did not collect it.
Error: Two
    at Promise (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:18:9)
    at new Promise (<anonymous>)
    at Object.<anonymous> (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:17:10)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:241:19) Promise {
  <rejected> Error: Two
    at Promise (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:18:9)
    at new Promise (<anonymous>)
    at Object.<anonymous> (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:17:10)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:241:19) }
Error: Three
    at Object.<anonymous> (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:21:27)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:241:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:577:3) Promise {
  <rejected> Error: Three
    at Object.<anonymous> (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:21:27)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:241:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:577:3) }
Second GC did not collect it.
Exit code 1
undefined:0



Error: Two
    at Promise (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:18:9)
    at new Promise (<anonymous>)
    at Object.<anonymous> (/Users/jgilli/dev/node-ruben-pr/test/message/promise_unhandled_reject.js:17:10)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:241:19)
$ echo $?
1
$ ls /cores/
$ 
```

Am I missing something? If not, is this the intended behavior?

Reply:
Based on the docs (unhandled rejections that are garbage collected are
raised as uncaught exceptions) it seems like the behavior is incorrect and `ERROR_ON_GC` should generate a core dump when used with `--abort-on-uncaught-exception`

Reply:
I agree and I'll fix this. @misterdjules thanks for bringing this up by the way :+1: I did not think about this case before.